### DISCUSSION TITLE
Ethernet W5100 SPI compatible ?

### DISCUSSION THREAD

**Discussion by Renber78** [sle118/squeezelite-esp32#299] [Q&A] [answered]
Do you think it is possible to run a Wiznet W5100 with the W5500 driver?

**Comment by sle118** [comment]
This is very unlikely to work with the existing driver. However, there's a proof of concept for a driver that works on the ESP-IDF, but until it is part of the official ESP-IDF framework we won't be able to support it.  

It is probablt doable, but would require a bit of work. Here's the driver: https://github.com/KaeLL/ESP32-W5100

**Comment by Renber78** [comment, marked as answer]
Thank you for your reply. I will then buy a W5500

**Comment by Renber78** [comment]
I installed my W5500 on the SPI port shared with my ST7789 display. It works but I don't know how to increase the speed of the W5500 to more than 20Mhz.
However, I used the dedicated port (VSPI). From the readme, I understand that it is possible to go up to 80Mhz.

eth_config: `model=w5500,cs=5,speed=40000000,intr=15`
spi_config: `mosi=23,clk=18,dc=14,miso=19`

`I (3222) network_ethernet: Detected driver w5500
I (3222) network_ethernet: Attempting to initialize Ethernet
I (3232) network_ethernet: Detected driver w5500
I (3232) network_ethernet: Ethernet config => model: w5500, valid: YES, type: SPI, mdc:0, mdio:0, rst:-1, intr:15, cs:5, speed:40000000, host:1
E (3242) spi_hal: spi_hal_cal_clock_conf(101): When work in full-duplex mode at frequency > 26.7MHz, device cannot read correct data.
Try to use IOMUX pins to increase the frequency limit, or use the half duplex mode.
Please note the SPI master can only work at divisors of 80MHz, and the driver always tries to find the closest frequency to your configuration.
Specify ``SPI_DEVICE_NO_DUMMY`` to ignore this checking. Then you can output data at higher speed, or read data at your own risk.
E (3292) spi_master: spi_bus_add_device(368): assigned clock speed not supported
E (3302) network_ethernet: SPI host failed : ESP_ERR_NOT_SUPPORTED
E (3302) messaging: Configuring Ethernet failed: ESP_ERR_NOT_SUPPORTED`

Also, I didn't understand putting 'host: -1' in the SPI config but I don't see any change by setting it to 1/2 or -1

Can you tell me if I'm going in the right direction?

**Comment by Renber78** [comment]
Good morning,

sle118, I saw that you are working on a 'SqueezeIO' PCB and that you are using this chip. Where is your project which looked promising?
Do you have a link for the diagram?
How did you wire the W5500 and what bus speed are you using?

THANKS
